tests/v2/validation/snapshot/snapshot.go

```diff
@@ -178,9 +167,9 @@ func snapshotRestoreRKE1(t *testing.T, client *rancher.Client, podTemplate corev
 
 		clusterResp.RancherKubernetesEngineConfig.Version = etcdRestore.UpgradeKubernetesVersion
 
-		if strategy {
-			clusterResp.RancherKubernetesEngineConfig.UpgradeStrategy.MaxUnavailableControlplane = cpUnavailalbleValue
-			clusterResp.RancherKubernetesEngineConfig.UpgradeStrategy.MaxUnavailableWorker = workerUnavailalbleValue
+		if etcdRestore.ControlPlaneUnavailableValue != "" && etcdRestore.WorkerUnavailableValue != "" {
+			clusterResp.RancherKubernetesEngineConfig.UpgradeStrategy.MaxUnavailableControlplane = etcdRestore.ControlPlaneUnavailableValue
+			clusterResp.RancherKubernetesEngineConfig.UpgradeStrategy.MaxUnavailableWorker = etcdRestore.WorkerUnavailableValue
```

Comment:
looking at this again after seeing the logs from jenkins, I think we should:
* add validation that these upgrade strategies are being respected. I realize that wasn't in before, but now that we're fleshing it out I think it makes sense to add a validation that the upgrade strategy is actually respected. would it be possible to do this, or do we depend on v3 to fail if it is not respected? 
* add logs to the jenkins run stating that the upgrade strategy is applied. Thoughts on this? looking at the logs, i wasn't able to tell that an upgrade strategy was used. Might be good to have.

Reply:
For the first point, there are require.Equal checks that validate that the user config upgrade strategy is matching what the cluster is currently at. The same is true for when upgrading Kubernetes version in the cluster.

For the second point, yeah I can add some logs to that, that's a good idea.